### PR TITLE
Update to latest dskit (new flagext.Secret and move of pkg/tenant)

### PR DESCRIPTION
**Pull request by simonswine**
- Update dskit
- Refactor to make use of new secret flag type
- Use tenant package from dskit

### CODE REVIEW

File: tempodb/backend/s3/config.go [hunk 1]
```diff
@@ -10,7 +10,7 @@ type Config struct {
 	Bucket            string         `yaml:"bucket"`
 	Endpoint          string         `yaml:"endpoint"`
 	Region            string         `yaml:"region"`
-	AccessKey         flagext.Secret `yaml:"access_key"`
+	AccessKey         string         `yaml:"access_key"`
```
Comment (annanay25) [inline tempodb/backend/s3/config.go:13]: I believe this change should be mentioned in the changelog as its a user facing change. The YAML marshaller will no longer print `AccessKey` as `********`

File: go.mod [hunk 2]
```diff
@@ -89,6 +89,8 @@ require (
 	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
 )
 
+require github.com/cespare/xxhash/v2 v2.1.2
```
Comment (annanay25) [inline go.mod:92]: nit: Can we club this require statement with the above block?

**Review by annanay25 — APPROVED** [review]
LGTM, thanks @simonswine. Left a few minor comments.

File: go.mod (hunk 2 above)
Comment (simonswine) [reply go.mod]: Unsure why that happened, now moved to the first require block

File: tempodb/backend/s3/config.go (hunk 1 above)
Comment (simonswine) [reply tempodb/backend/s3/config.go:13]: Good point added a changelog entry

**Comment by simonswine** [comment]
A bit unsure what's wrong with CI, but it seems to be in other PRs as well: (e.g. https://github.com/grafana/tempo/runs/5822436282?check_suite_focus=true)

**Review by annanay25 — APPROVED** [review]
Thanks! LGTM.

**Pull request merged**